app/src/main/java/org/bspb/smartbirds/pro/ui/MonitoringActivity.java

```diff
@@ -181,6 +181,29 @@
     private Set<String> formsEnabled;
     private Menu menu;
 
+    private ServiceConnection trackingServiceConnection = new ServiceConnection() {
+        @Override
+        public void onServiceConnected(ComponentName name, IBinder service) {
+            Log.d(TAG, String.format(Locale.ENGLISH, "service %s connected", name));
+        }
+
+        @Override
+        public void onServiceDisconnected(ComponentName name) {
+            Log.d(TAG, String.format(Locale.ENGLISH, "service %s disconnected", name));
+        }
+    };
+
+    private ServiceConnection dataServiceConnection = new ServiceConnection() {
+        @Override
+        public void onServiceConnected(ComponentName name, IBinder service) {
+            Log.d(TAG, String.format(Locale.ENGLISH, "service %s connected", name));
+        }
+
+        @Override
+        public void onServiceDisconnected(ComponentName name) {
+            Log.d(TAG, String.format(Locale.ENGLISH, "service %s disconnected", name));
```

Comment:
This should happen only when destroying so we can check if `Activity.this.isFinishing` and if not to log exception and finish.